packages/coingecko/src/__tests__/e2e/integration.spec.ts

```diff
@@ -79,4 +78,34 @@ describe("Coingecko", () => {
     expect(result.data?.tokenMarketChart.market_caps[0]).toBeTruthy();
     expect(result.data?.tokenMarketChart.total_volumes[0]).toBeTruthy();
   });
+
+  it("should get coin market chart data for a range", async () => {
+    const id = "ethereum";
+    const vsCurrency = "usd";
+    const from = 1638463900;
+    const to = 1638465000;
+
+    const result = await client.query<CoinMarketChartRangeResult>({
+      uri: ensUri,
+      query: `
+        query($id: String!, $vsCurrency: String!, $from: UInt!, $to: UInt!) {
+          coinMarketChartRange(id: $id, vs_currency: $vsCurrency, from: $from, to: $to)
+        }
+      `,
+      variables: { id, vsCurrency, from, to },
+    });
+
+    expect(result.errors).toBeFalsy();
+    expect(result.data).toBeTruthy();
+
+    // check presence of properties
+    expect(result.data?.coinMarketChartRange).toHaveProperty("prices");
+    expect(result.data?.coinMarketChartRange).toHaveProperty("market_caps");
+    expect(result.data?.coinMarketChartRange).toHaveProperty("total_volumes");
+
+    // check one item for each prop
+    expect(result.data?.coinMarketChartRange.prices[0]).toBeTruthy();
```

Comment:
Let's also add test for checking length of these arrays.
```
    expect(result.data?.coinMarketChartRange.prices.length).toBe(4);
```

Reply:
Thanks for your feedback, I've added the test cases.